File: 11448-Ga/main.cpp
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

int dx[8] = {-1, 0, 1, 1, 1, 0, -1, -1};
int dy[8] = {-1, -1, -1, 0, 1, 1, 1, 0};

int N, wcount;
queue<pair<int, int>> q;

void bfs(vector<vector<char>>& board, vector<vector<bool>>& visited, int x,
         int y) {
  if (x < 0 || y < 0 || x >= N || y >= N) return;
  if (visited[y][x]) return;

  visited[y][x] = true;
  if (board[y][x] == '-') {
    wcount++;
    q.push(make_pair(x, y));
  }
}

int main() {
  int T, queue_size;
  pair<int, int> pos;
  int x, y;
  cin >> T;

  for (int i = 0; i < T; i++) {
    cin >> N;
    wcount = 0;

    vector<vector<char>> board(N, vector<char>(N));
    vector<vector<bool>> visited(N, vector<bool>(N, false));
    for (int j = 0; j < N; j++)
      for (int k = 0; k < N; k++) cin >> board[j][k];

    for (int j = 0; j < N; j++) {
      for (int k = 0; k < N; k++) {
        if (board[j][k] == 'w') {
          visited[j][k] = true;
          q.push(make_pair(k, j));

          while (q.size() != 0) {
            queue_size = q.size();

            while (queue_size > 0) {
              pos = q.front();
              q.pop();
              x = pos.first;
              y = pos.second;

              for (int l = 0; l < 8; l++)
                bfs(board, visited, x + dx[l], y + dy[l]);

              queue_size--;
            }
          }
        }
      }
    }

    cout << wcount << endl;
  }
}

File: 11448-Ga/main.py
import sys
input = sys.stdin.readline;

dx = [-1, 0, 1, 1, 1, 0, -1, -1]
dy = [-1, -1, -1, 0, 1, 1, 1, 0]

T = int(input())
N = 0
count = 0
q = []

def bfs(board: list[str], visited: list[list[bool]], x: int, y: int):
  if x < 0 or y < 0 or x >= N or y >= N:
    return
  if visited[y][x]:
    return
  
  visited[y][x] = True
  if board[y][x] == "-":
    global count
    count += 1
    q.append([x, y])

for i in range(T):
  N = int(input())
  count = 0

  board = []
  visited = [[False] * N for _ in range(N)]

  for _ in range(N):
    board.append(input())

  for j in range(N):
    for k in range(N):
      if board[j][k] == "w":
        visited[j][k] = True
        q.append([k, j])

        while len(q) != 0:
          queue_size = len(q)

          while queue_size > 0:
            x, y = q.pop()

            for m in range(8):
              bfs(board, visited, x + dx[m], y + dy[m])

            queue_size -= 1

  print(count)